Plateau degeu

QuentinRimet opened the pull request:
- création d'un plateau de jeu et modélisation de celui-ci en carré
- chaque thème possède sa propre couleur
- ajout de carré de couleur pour les joueurs
- 4 joueurs par plateau
- utilisation de twig
- route définie dans l'index
- suivant le modèle mvc

Lilychaan committed:
ajout joueur sur plateau

QuentinRimet committed:
Creation et affichage d'un plateau via un lien

QuentinRimet committed:
Merge branch 'master' into Plateau-Degeu

ElGitariste commented:
Ok ça à l'air bon !

QuentinRimet committed:
Merge branch 'master' into Plateau-Degeu

floo51 commented on index.php:
```diff
@@ -40,6 +40,13 @@
 
+
+$app->get('/plateau', function() {
+	$acc = new jeuControlleur();
```
Naming should be in english.

floo51 commented on index.php:
```diff
@@ -40,6 +40,13 @@
+$app->get('/plateau', function() {
+	$acc = new jeuControlleur();
+	$acc->initJeu();
+	$acc->affichageJeu();
```
A method should start with a verb, not a noun. It's an action, not an object.

floo51 commented on src/Jeu.php:
```diff
@@ -0,0 +1,17 @@
+        $this->plateau->creaPlateau();
+    }
+
+    public function getPlateau(){
```
Is there a point having a getter instead of making your `$plateau` property public?

floo51 commented on src/Plateau.php:
```diff
@@ -1,137 +1,92 @@
-        else
-        if($i==1 || ($i-0.5==$taille_tab/2 && $j<$taille_tab/2)){
+    public function creaPlateau(){
+    for($i=1;$i<=$this->taille_tab;$i++){
```
This is hardly readable, some clues:
- Can you write this code in another way? Maybe just initialize the table without any for loops (the table doesn't change size right?)
- Can you try to externalize some of the code to named functions?
- Can you take more care with variable naming? `$i` and `$j` usually can be replaced with things like `$column` and `$row`

floo51 commented on src/Plateau.php:
```diff
@@ -1,137 +1,92 @@
+            else
+            if($i==1 || ($i-0.5==$this->taille_tab/2 && $j<$this->taille_tab/2)){
+                switch($j%6){
+                    case 0:$this->tableau[$i][$j]=["theme"=>"geo","player"=>array()];
```
`"geo", "diver", ...` are called magic strings and are hell for maintainability. Use constants instead?

floo51 commented on src/Plateau.php:
```diff
@@ -1,137 +1,92 @@
-        else
-         $tableau[$i][$j]="";
+
+    function getTableau(){
```
Is there a point having a getter instead of a public property?

floo51 commented on src/Plateau.php:
```diff
@@ -1,137 +1,92 @@
-        
-        echo "</td>";
+    function getTaille(){
+        return $this->taille_tab;
```
Does the game board size actually change? If not, why use a variable?

floo51 commented on src/vues/JeuVue.php:
```diff
@@ -0,0 +1,100 @@
+
+class JeuVue {
+
+    public function renderPlateau($plateau){
```
This is not OK. You can't have a templating engine AND generate html with string concatenation.

floo51 commented on src/vues/JeuVue.php:
```diff
@@ -0,0 +1,100 @@
+        }
+
+	public function render($a) {
+        $ach = GlobaleVue::header();
```
Be careful with variable naming, I have no idea what `$ach` and `$acf` refer to.

QuentinRimet committed:
modification affichage plateau

-passage a twig
-renommage en anglais
-création du tableau simplifié
-adaptation slim 3

QuentinRimet committed:
passage attribut en public

floo51 commented on container.php:
```diff
@@ -0,0 +1,27 @@
+];
+$container = new \Slim\Container($configuration);
+$container['GameController'] = function($c) {
+	$view = $c->get("view"); // retrieve the 'view' from the container
```
Not sure the comment is needed there, the code explains itself.

floo51 commented on container.php:
```diff
@@ -0,0 +1,27 @@
+$container['view'] = function ($container) {
+    $view = new \Slim\Views\Twig('src/vues');
+
+    // Instantiate and add Slim specific extension
```
This one seems much more needed because I'm not able to understand the 2 lines below 👍

floo51 commented on src/Board.php:
```diff
@@ -0,0 +1,94 @@
+    public $grid = array();
+
+    public function newBoard(){
+    $hist="hist";
```
That's variables, not constants: https://stackoverflow.com/questions/2447791/define-vs-const

floo51 commented on src/Board.php:
```diff
@@ -0,0 +1,94 @@
+    $info="info";
+    $perso="perso"; 
+
+    $this->grid[1][1]=["theme"=>$hist,"player"=>array()];
```
This is, to me, much more readable than all the loops and switches 👍

floo51 commented on src/vues/GameView.html.twig:
```diff
@@ -0,0 +1,21 @@
+{% include 'GlobalHeaderView.html.twig' %}
+<table>
```
Not mandatory for this review, you might want to use more responsive html tags. `div` with css grid by instance: https://css-tricks.com/snippets/css/complete-guide-grid/

floo51 commented on src/vues/GlobalHeaderView.html.twig:
```diff
@@ -0,0 +1,10 @@
+    {% block stylesheets %}
+        <link rel="stylesheet" href="web/css/board.css"/>
+    {% endblock %}
+    </head>
```
Seems like there are missing closing tags?

QuentinRimet committed:
reponsive html + some change

-constante
-commentaire

QuentinRimet committed:
correction erreur placement balise
